doc/architecture/computing-services.md

```diff
@@ -0,0 +1,105 @@
+  - Regulation
+
+* Storage services for new derived assets
+
+# Architecture
+
+## Enabling Publisher Services (Brizo)
+
+The direct interaction with the infrastructure where the data is, requires the execution of a component handled by Publishers.
+This component will be in charge of interact with the users, and manage the basics of own Publisher infrastructure to provide the additional services.
+This business logic supporting the some additional Publisher capabilities, is responsibility of a new technical component.
+
+The main & new key component introduced to support new Publisher services is named **Brizo**.
+
+_"Brizo is an ancient Greek goddess who was known as the protector of mariners, sailors, and fishermen. She was worshipped primarily by the women of Delos, who set out food offerings in small boats. Brizo was also known as a prophet specializing in the interpretation of dreams."_
+
+In the "Ocean ecosystem", Brizo is the technical component executed by the **Publishers** allowing to them to provide extended data services.
+Brizo, as part of the Publisher ecosystem, includes the credentials to interact with the infrastructure (initially cloud, but could be on-premise).
+Because of those credentials, the execution of Brizo **SHOULD NOT** be delegated to a third-party.
+
+![Brizo High-Level Architecture](img/brizo-hl-arch.png)
+
+
+## Responsibilities
+
+The main responsibilities of the software are:
+
+* Expose a HTTP api allowing to execute some extended data services (compute and/or storage)
+* Authorize the user on-chain using the proper Service Agreement. It allows to validate that user requesting the service is allowed to use that service.
+* Interact with the infrastructure (cloud/on-premise) using the Publisher credentials.
+* Start/stop/execute computing instances with the algorithm's provided by the users
+* Retrieve the logs generated during the executions
+* Register new Assets derivated of the executions as new Ocean Assets
+* Provide **Proofs of Computation** to the Ocean Network
+
+## Initial solution
+
+![Sequence Diagram for computing services](img/computing-service-flow.png)
+
+In the above diagram you can see the initial integration supported. It involves the following components/actors:
+
+* Datascientists - Final users requiring to use some computing/storage services where the data they are using is.
+* Ocean Keeper - In charge of maintain the Service Agreements and supporting the on-chain authorization
+* Publisher Infrastrucure - Cloud or On-Premise. Is were the Publisher data is.
+* Publisher Agent (Brizo) - Orchestrate/provides the additional publisher services
+* Publisher - Is the user owning the data and running Brizo. He/she doesn't interact actively in this flow.
+
+The pre-conditions of this flow are:
+
+* In a **previous stage** to this flow, the datascientist got access to an Asset (dataset) thru a classical purchase process
+* The datascientist owns an Asset of type Algorithm registered in Ocean
```

Comment:
Does the data scientist have to _own_ the Algorithm, or do they just have to have the right to use it?